### ISSUE TITLE
CMake does not detect fancy c++ compiler (which knows the auto keyword)

### ISSUE THREAD

**Issue by wdoekes** [SIPp/sipp#486]
See #485 

According to reporter, cmake will happily choose an old c++ compiler (from /bin/c++, apparently).

If we're using fancy new c++ stuff, it'd be nice to have cmake check if the compiler supports it before starting any build at all.

**Comment by nchaigne** [comment]
As commented in https://github.com/SIPp/sipp/issues/485:

CMake does correctly use the newer compiler, however the compiler must be installed before running "./build.sh" (CMake files are not re-generated).

**Comment by wdoekes** [comment]
Ok. So 626de652455a6c061d5a045fb226212c1f7eaeb5 made things work with pre-C++11 compilers.

And for the dev (master or 3.7.x) branch I added the appropriate checks.

Thanks for the report!
